v2/contribute/decisions/accountlinking/0001-support-for-header-based-auth.mdx

```diff
@@ -0,0 +1,31 @@
+---
+id: "0001"
+title: Recipe functions should take recipeUserId or primaryUserId
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Recipe functions should take recipeUserId or primaryUserId
+
+<DecisionHeader status="accepted" lastUpdate="2022-11-23" created="2022-11-23" deciders={["rishabhpoddar", "bhumilsarvaiya"]} proposedBy={["bhumilsarvaiya"]} />
+
+## Context and Problem Statement
+
+There are a few recipe functions that take userId and it can get confusing at times to decide whether the userId passed should be treated as primaryUserId or recipeUserId
+
+## Example
+
+For `updateEmailOrPassword` function, let's say the input are userId and password. If the passed userId is recipeUserId, it's all fine. But what if the userId is primaryUserId and there are multiple emailpassword users linked with the primaryUserId. For which one should we update the password?
+
+## Considered Options
+
+* Have recipeUserId and primaryUserId everywhere (no id). So functions like getUserId in session object will no longer be there. Instead, there will be getPrimaryUserId and getRecipeUserId (this will be there anyways).
+* For the passed userId if there exists a recipeUserId, we apply changes to that recipeUserId. If recipUserId doesn't exists but there exists a primaryUserId, we try to get recipeUserId on which the changes can be applied. If no recipe user is found, we simply throw no recipe user found error. If more than one recipe user is found for which the changes can be applied, we throw an error indicating to the user to pass a recipeUserId instead.
+
```

Comment:
There is also one more option we spoke about -> these functions should only accept a recipeUserId and if a primaryUserId is passed, it always throws an error.